posts/2022-03-15-ga-release-post.adoc

```diff
@@ -0,0 +1,154 @@
+
+When Liberty detects a failure to write to its transaction recovery logs, it invalidates the logs and disallows further global transactions. To resume transactional work, you have to restart the server and reload any applications, which can take considerable time in some cases.
+
+Under certain circumstances, such as when the recovery log failure is caused by transient database connection errors, the condition can be handled by retrying the database operation thus avoiding the need to shutdown the server and reload the apps.
+
+A new attribute is available for the `transaction` configuration element in the Liberty `server.xml` file that enables these transaction log SQL operations to be retried. It is named `enableLogRetries` and should be set to `true` to enable retries. For example,
+
+[source,xml]
+----
+<transaction
+dataSourceRef="tranlogDataSource"
+recoverOnStartup="true"
+waitForRecovery="false"
+heuristicRetryInterval="10"
+enableLogRetries="true"
+/>
+----
+
+By default the new functionality is disabled.
+
+For more information on the storing or Transaction Recovery Logs in an `RDBMS`, please refer to link:{url-prefix}/docs/latest/transaction-service.html#log[Transaction log management] doc.
+
+
+[#bugs]
+== Notable bugs fixed in this release
+
+We’ve spent some time fixing bugs. The following sections describe just some of the issues resolved in this release. If you’re interested, here’s the  link:https://github.com/OpenLiberty/open-liberty/issues?q=label%3Arelease%3A22003+label%3A%22release+bug%22+[full list of bugs fixed in 22.0.0.3].
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/20165[jsonpContainer-2.0 and jsonbContainer-2.0 features incorrectly use default providers]
++
+The `jsonpContainer-2.0` and `jsonbContainer-2.0` features incorrectly used default providers. These features have been updated to include the BELLS feature and remove Liberty's default implementations of JSON-P and JSON-B. This allows for a user to include their own third party library for these JSON technologies. Additionally this fix updates the `JSON` container `FAT` tests to run against the 2.0 json container features (Jakarta EE 9) to verify this fixed functionality.
+
+* link:https://github.com/OpenLiberty/open-liberty/issues/12050[@RolesAllowed rejects unauthenticated users when they mapped to an allowed (EVERYONE) role]
++
+Previously, the special-subject `EVERYONE` did not play well with the `@RolesAllowed` annotation on `JAX-RS` resources. When the special-subject `EVERYONE` is bound to a `security-role`, unauthenticated users should be considered to be in that role and should have access to the endpoints allowed for that role. The `@RolesAllowed` annotation would reject unauthenticated users when they were mapped to an allowed (`EVERYONE`) role. This issue has since been resolved.
```

Comment:
```suggestion
Previously, the special-subject `EVERYONE` did not work correctly with the `@RolesAllowed` annotation on `JAX-RS` resources. When the special-subject `EVERYONE` is bound to a `security-role`, unauthenticated users should be considered to be in that role and should have access to the endpoints allowed for that role. The `@RolesAllowed` annotation would reject unauthenticated users when they were mapped to an allowed (`EVERYONE`) role. This issue has since been resolved.
```